### ISSUE TITLE
Feature request: Ask for highlighting to be turned on for really long documents

### ISSUE THREAD

**Issue by Piiit** [linuxmint/xed#142]
Sometimes I must open huge text files, more than 1M lines (csv-files or a bunch of sql-inserts for instance). They open now quite fast, within a minute showing a progress bar. However, afterwards the hightlighting starts and xed gets unresponsible for a long time. Would be nice to have syntax hightlighting turned off by default for long files (more than 100k lines?), and show a warning (similar to the progress bar) to ask if hightlighting should be turned on, and that it may take some time...

**Comment by JosephMcc** [comment]
Last time I tried this, you could simply set it to plain text while it's loading and that fixes the problem. If you have a file that doesn't work with, could you share it? You know how to get a hold of me if you want to do it privately ;)

**Comment by Piiit** [comment]
@JosephMcc Yes this works. However, nevertheless this would be a nice feature, especially for users that do not know why it is lagging.

**Comment by Piiit** [comment]
*EDIT* for my previous comment: 

It does only work if the non-highlighting (regular text) was set before. It gets stored as gvfs metadata. `gvfs-info` gives `metadata::xed-language: _NORMAL_`, when hightlighting was turned off. However, when a file gets opened the first time, and during the load process set to "regular text" for no hightlighting, after loading its set to the language detected.
